File: pams-backend/src/main/java/com/pams/module/content/entity/News.java
package com.pams.module.content.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.SQLRestriction;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "news")
@SQLRestriction("deleted = 0")
public class News {
    public enum NewsStatus { DRAFT, PUBLISHED }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, length = 150)
    private String title;
    @Column(length = 300)
    private String subtitle;
    @Column(columnDefinition = "TEXT")
    private String content;
    private Long activityId;
    private Long authorId;
    private LocalDate publishDate;
    @Enumerated(EnumType.STRING)
    private NewsStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Integer deleted;
}

File: pams-web/src/api/news.ts
import { get, post, put, del } from './http'
import type { PageResult } from './types'

export interface NewsVO {
  id: number
  title: string
  subtitle: string
  content: string
  activityId: number | null
  authorId: number | null
  publishDate: string | null
  status: 'DRAFT' | 'PUBLISHED'
  createdAt: string | null
  updatedAt: string | null
}

export interface NewsSave {
  title: string
  subtitle?: string
  content?: string
  activityId?: number | null
}

export const NEWS_STATUS_MAP: Record<string, string> = {
  DRAFT: '草稿',
  PUBLISHED: '已发布',
}

export const NEWS_STATUS_OPTIONS = Object.entries(NEWS_STATUS_MAP).map(([value, label]) => ({ value, label }))

export const listNews = (params: { keyword?: string; page?: number; size?: number }) =>
  get<PageResult<NewsVO>>('/news', params)
export const createNews = (data: NewsSave) => post<number>('/news', data)
export const updateNews = (id: number, data: NewsSave) => put<void>(`/news/${id}`, data)
export const deleteNews = (id: number) => del<void>(`/news/${id}`)
